File: ai25btech11033/matgeo/matgeo-2.8.7/codes/main..c
#include <stdio.h>
#include <math.h>

int main() {
    // Given values
    double a = 2.0;   // |a|
    double b = 3.0;   // |b|
    double dot = 4.0; // a · b

    // Compute cos(theta)
    double cos_theta = dot / (a * b);

    // Compute angle in radians
    double theta_rad = acos(cos_theta);

    // Convert to degrees
    double theta_deg = theta_rad * (180.0 / M_PI);

    // Print results
    printf("cos(theta) = %.2f\n", cos_theta);
    printf("theta (radians) = %.4f\n", theta_rad);
    printf("theta (degrees) = %.2f\n", theta_deg);

    return 0;
}

File: ai25btech11033/matgeo/matgeo-2.8.7/codes/main.py
import math

# Given values
a = 2.0   # |a|
b = 3.0   # |b|
dot = 4.0 # a · b

# Compute cos(theta)
cos_theta = dot / (a * b)

# Compute angle in radians
theta_rad = math.acos(cos_theta)

# Convert to degrees
theta_deg = math.degrees(theta_rad)

# Print results
print("cos(theta) =", round(cos_theta, 2))
print("theta (radians) =", round(theta_rad, 4))
print("theta (degrees) =", round(theta_deg, 2))
